### DISCUSSION TITLE
split up large xml files with indexes for manual create

### DISCUSSION THREAD

**Discussion by gottaloveit** [spatie/laravel-sitemap#459] [Ideas]
So, following the readme, I added this to my Video model:
```
    public function toSitemapTag(): Url | string | array
    {
        return route('video', $this);
    }
```

And I create the sitemap like so, since I do not want it to crawl because I got over 7000 videos, so the database query option is obviously better and most accurate.
```
Sitemap::create()->add(Video::all())->writeToDisk('public', 'sitemap.xml');
```

Problem though is it creates it in a single file and everything I have read says to split up large files.  So I try:
```
Sitemap::create()->maxTagsPerSitemap(1000)->add(Video::all())->writeToDisk('public', 'sitemap.xml');
```
I get error: 

> Call to undefined method Spatie\Sitemap\Sitemap::maxTagsPerSitemap() 

I can't use SitemapGenerator because the create method requires a url
```
SitemapGenerator::create()->maxTagsPerSitemap(1000)->add(Video::all())->writeToDisk('public', 'sitemap.xml');
```

> Too few arguments to function Spatie\Sitemap\SitemapGenerator::create()

So how can I use the maxTagsPerSitemap method and get indexes when creating manually?

Thanks

**Comment by theshubhamchugh** [comment]
+1
